docs/dev/checks/README.md

```diff
@@ -133,3 +125,91 @@ checks code.
+
+## Running checks with a local Agent Build
+### Custom Checks
+Scenario: You have implemented a custom check called `hello_world` and you'd like
+to run this with a local agent build.
+
+1. Place the configuration file `hello_world.yaml` in `dev/dist/conf.d/` folder
+2. Place the pthon code in `dev/dist/` folder
+2. Run `inv agent.build` as usual. This step will copy the contents
+   of `dev/dist` into `bin/agent/dist`, which is where the agent will be
+   able to find your code.
+
+The resulting directory structure should look like:
+```
+dev/dist
+├── conf.d
+│   └── hello_world.yaml
+├── hello_world.py
+└── README.md
+```
+
+### Standard Checks
+There are a number of checks that ship with a full build of the Agent, but when
+you're developing against a local build, you will not get any of these python
+checks out of the box, you must install them.
+
+> Note the following instructions install python packages to the python user
+> install directory. This could conflict with existing packages.
+
+The list of checks currently shipping with the agent lives in the
+[integrations-core repo
+here](https://github.com/DataDog/integrations-core/blob/master/requirements-agent-release.txt)
+
+Say you want to test the `openmetrics` check.
+
+1. Clone `integrations-core` locally. (Optionally checkout the git tag
+   corresponding to the version you want to test)
+2. From integration-core directory, install `datadog-checks-base` as a pre-req.
+   `pip3.10 install --user './datadog_checks_base[deps]'`
+3. Install the `openmetrics` check. From inside the `integrations-core`
+   checkout:
+   `pip3.10 install --user ./openmetrics`
+
+That's it! Your local build should now have the correct packages to be able to
+run the `openmetrics` check.
+
+#### "What is this `[deps]` thing?"
+The `[deps]` at the end of the package name instructs pip to install the
+"requires" that match the `deps` "extra". Without this, you'll see errors when running the agent
+about missing dependencies.
+
+View all the possible 'extras' with:
+```
+pip3.10 install --no-warn-script-location --user --dry-run --ignore-installed -qqq --report - datadog-checks-base | jq '.install[] | select(.metadata.name=="datadog-checks-base") | .metadata.provides_extra'
+```
+
+View all the "requires" (which shows which "extra" will trigger that
+dependency to be installed) with:
+```
+pip3.10 install --no-warn-script-location --user --dry-run --ignore-installed -qqq --report - datadog-checks-base | jq '.install[] | select(.metadata.name=="datadog-checks-base") | .metadata.requires_dist'
+```
+
+#### "Which python is my agent build using?"
+Assuming you're not doing a full omnibus build, you won't have an embedded
+python in your local build, so your local build will need to choose a python
+from your system to use.
+
+By default the agent will choose the first python binary (`python` or `python3`
+depending on your configuration) from the `PATH` that is used to run the agent.
```

Comment:
It's a bit more subtle than that. If I remember right, we load python shared lib via rtloader that then loads the default python based on some env vars I think but that may be overkill to explain in this doc.

Reply:
I'll include a brief note about this, it seems the code checks for `PythonHome` and uses that if its set, but I didn't get around to testing this part.